Clarify CA1860 and CA1827 (#36257)
dotnet/docs: docs/fundamentals/code-analysis/quality-rules/ca1827.md

## Cause

> The <xref:System.Linq.Enumerable.Count%2A>() or <xref:System.Linq.Enumerable.LongCount%2A>() **method** was used where the <xref:System.Linq.Enumerable.Any%2A>() method would be more efficient.

Reviewer: ```suggestion
The [Count()](xref:System.Linq.Enumerable.Count%2A) or [LongCount()](xref:System.Linq.Enumerable.LongCount%2A) *method* was used where the [Any()](<xref:System.Linq.Enumerable.Any%2A) method would be more efficient.
```

## Rule description

> This rule flags the <xref:System.Linq.Enumerable.Count%2A>() and <xref:System.Linq.Enumerable.LongCount%2A>() LINQ method calls used to check if the collection has at least one element. These method calls require enumerating the entire collection to compute the count. The same check is faster with the <xref:System.Linq.Enumerable.Any%2A>() method as it avoids enumerating the collection.

Reviewer: ```suggestion
This rule flags [Count()](xref:System.Linq.Enumerable.Count%2A) and [LongCount()](xref:System.Linq.Enumerable.LongCount%2A) LINQ method calls that are used to check if the collection has at least one element. These methods enumerate the entire collection to compute the count. The same check is faster with the [Any()](<xref:System.Linq.Enumerable.Any%2A) method as it avoids enumerating the collection.
```

}
```

> [!NOTE]

> > This rule is similar to [CA1860: Avoid using 'Enumerable.Any()' extension method](ca1860.md). However that rule suggests using the `Count` **property**, while this rule applies to the Linq `Count()` extension **method**.

Reviewer: ```suggestion
> This rule is similar to [CA1860: Avoid using 'Enumerable.Any()' extension method](ca1860.md). However that rule suggests using the `Count` *property*, while this rule applies to the Linq `Count()` *extension method*.
```
